client/cgo/verification.go

```diff
@@ -55,13 +56,15 @@ func cgoVerifyVrf(pk unsafe.Pointer, pkSize C.int,
 }
 
 //export cgoVerifyHashChain
-func cgoVerifyHashChain(prevHash unsafe.Pointer, root unsafe.Pointer, size C.int) C.int {
-	if int(size) != crypto.HashSizeByte {
+func cgoVerifyHashChain(prevHash unsafe.Pointer, hashSize C.int, strSig unsafe.Pointer, sigSize C.int) C.int {
```

Comment:
I might rename ``strSig`` to ``prevStrSig`` (and maybe even ``prevHash`` to ``curPrevStrHash``) just to make it absolutely clear what it is that you're comparing. At first glance, it seems like you might be comparing the current ``prevStrHash`` with the current STR's signature.